wg-api/spec-documents/scroll-view.md

```diff
@@ -0,0 +1,178 @@
+
+const window = new BrowserWindow({ width: 1000, height: 500 })
+
+const scroll = new ScrollView()
+scroll.setBounds({ width: 1000, height: 500 })
+
+const scrollContent = new ContainerView()
+scrollContent.setBounds({ width: 2000, height: 500 })
+
+const browserView1 = new BrowserView()
+const wrapperBrowserView1 = new WrapperBrowserView({ 'browserView': browserView1 });
+wrapperBrowserView1.setBounds({ width: 1000, height: 500 })
+scrollContent.addBrowserView(wrapperBrowserView1)
+
+const browserView2 = new BrowserView()
+const wrapperBrowserView2 = new WrapperBrowserView({ 'browserView': browserView2 });
+wrapperBrowserView2.setBounds({ width: 1000, height: 500, x: 1000 })
+scrollContent.addBrowserView(wrapperBrowserView2)
+
+scroll.setContentView(scrollContent);
+win.addChildView(scroll);
+```
+
+
+## API Design
+
+In order to achieve the example above, we've made following changes to the API
+
+### BaseView
+
+A `BaseView` is a rectangle within the views View hierarchy. It is the base
+class for all all different views. 
+
+Process: **Main**
+
+`BaseView` is an EventEmitter
+
+
+#### Static Methods
+
+* `BaseView.getAllViews()`
+* `BaseView.fromId(id)`
+
+#### Instance Properties
+
+* `id` - A `Integer` property representing the unique ID of the view. Each ID is unique among all `BaseView` instances of the entire Electron application.
+* `isContainer` (*Readonly, Experimental*) A `boolean` property that determines whether this view is (or inherits from) ContainerView.
+
+#### Instance Methods
+
+* `setBounds(bounds)`
+* `getBounds()`
+* `setBackgroundColor(color)`
+* `getParentView()` - Returns `BaseView || null` - The parent view, otherwise returns `null`.
+* `getParentWindow()` - Returns `BrowserWindow || null` - The window that the view belongs to, otherwise returns 	`null`
+
+	*Note: The view can belong to either a view or a window.* 
+
+#### Events
+
+* `size-changed`
+  
+- - - 
+
+### BrowserWindow 
+
+#### New instance methods
+
+* `addChildView(view)` 
+* `removeChildView(view)`
+* `getViews()`
+
+- - -
+
+### ContainerView
+
+A `ContainerView` can be used to embed additional views hierarchy into a `BrowserWindow`. It extends `BaseView`
+
+Process: **Main**
+
+#### Static Methods
+
+* `ContainerView.fromId(id)` 
+
+#### Instance Methods
+
+* `addChildView(view)`
+* `removeChildView(view)`
+* `getViews()` - Returns an array of all BaseViews that have been attached
+
+
+### ScrollView
+
+Show a part of view with scrollbar.  The `ScrollView` can show an arbitrary content view inside it. It is used to make any View scrollable. When the content is larger than the `ScrollView`, scrollbars will be optionally showed. When the content view is smaller then the `ScrollView`, the content view will be resized to the size of the `ScrollView`. The scrollview supports keyboard UI and mousewheel. It extends `BaseView`.
+
+Process: **Main**
+
+
+#### Static Methods
+
+
+* `ScrollView.fromId(id)`
+
+#### Instance Methods
```

Comment:
would you mind reformatting this section to match the format of our docs? It's a little hard to read as is.

Reply:
I reformated the API description, hopefully it's better now.